hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/CustomFixedFeeAssessor.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2020-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.handlers.transfer.customfees;
+
+import static com.hedera.node.app.service.token.impl.handlers.transfer.customfees.AdjustmentUtils.adjustHbarFees;
+import static com.hedera.node.app.service.token.impl.handlers.transfer.customfees.AdjustmentUtils.adjustHtsFees;
+import static com.hedera.node.app.service.token.impl.handlers.transfer.customfees.AdjustmentUtils.couldTriggerCustomFees;
+import static com.hedera.node.app.service.token.impl.handlers.transfer.customfees.CustomFeeExemptions.isPayerExempt;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.transaction.AssessedCustomFee;
+import com.hedera.hapi.node.transaction.CustomFee;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Assesses fixed fees in a custom fee.
+ * All the custom fees assessed from fixed fee will be added to the {@link AssessmentResult}
+ * which will be used to create next level of transaction body to assess the next level of custom fees.
+ * @see CustomFeeAssessor
+ */
+@Singleton
+public class CustomFixedFeeAssessor {
+    @Inject
+    public CustomFixedFeeAssessor() {
+        // For Dagger injection
+    }
+
+    /**
+     * Assesses fixed fees in a custom fee.
+     * @param feeMeta the custom fee metadata for the token
+     * @param sender the sender of the transaction, which will be payer for custom fees
+     * @param result the assessment result which will be used to create next level of transaction body
+     */
+    public void assessFixedFees(
```

Comment:
If we weren't in such a hurry to get things implemented, I would suggest that this method return a `CustomFeeMeta` object. One of the reasons I struggle with the mono service code is that mutable objects are continually passed around, sometimes many function calls deep, and manipulated in various scattered places. It's difficult to figure out the state of an object like this when reading code. I think fees would be much more manageable if we wrote them in such a way that the algorithm didn't change the `CustomFeeMeta` at all, and instead produced the result as independently as possible. 

That said, we're in a hurry, so this comment is essentially useless right now. But since I wrote it anyway... 🤷 😂